features/environment.py

```diff
@@ -139,11 +178,37 @@ def from_environ(cls, config) -> "UAClientBehaveConfig":
 def before_all(context: Context) -> None:
     """behave will invoke this before anything else happens."""
     userdata = context.config.userdata
+    if userdata:
+        print("Userdata key / value pairs:")
+        for key, value in context.config.userdata.items():
+            print("   - {} = {}".format(key, value))
     context.series_image_name = {}
     context.series_reuse_image = ""
     context.reuse_container = {}
     context.config = UAClientBehaveConfig.from_environ(context.config)
-
+    has_aws_keys = bool(
+        context.config.aws_access_key_id
+        and context.config.aws_secret_access_key
+    )
+    if has_aws_keys:
```

Comment:
If the `has_aws_keys` check fails, we will not create the `ec2_api` variable in `context.config`. That will end up causing the scenario to fail when we try to create a ec2 instance in the `given_a_machine` step. Maybe we could abort the test here if the check fails and we are running pro tests